src/main/java/players/rhea/RHEAPlayer.java

```diff
@@ -0,0 +1,333 @@
+            child.actions[child.length - 1 - i] = p2.actions[child.length - 1 - i];
+            child.gameStates[child.length - 1 - i] = p2.gameStates[child.length - 1 - i].copy();
+        }
+        return child;
+    }
+
+    RHEAIndividual[] selectParents()
+    {
+        RHEAIndividual[] parents = new RHEAIndividual[2];
+
+        switch (params.selectionType)
+        {
+            case TOURNAMENT:
+                parents[0] = tournamentSelection();
+                parents[1] = tournamentSelection();
+                break;
+            case RANK:
+                parents[0] = rankSelection();
+                parents[1] = rankSelection();
+                break;
+            default:
+                throw new RuntimeException("Unexpected selection type");
+        }
+
+        return parents;
+    }
+
+    RHEAIndividual tournamentSelection()
+    {
+        int rand = randomGenerator.nextInt(population.size() - params.tournamentSize);
+        RHEAIndividual best = population.get(rand);
+        for(int i = rand + 1; i < rand + params.tournamentSize; ++i)
+        {
+            RHEAIndividual current = population.get(i);
+            if(current.value > best.value)
+                best = current;
+        }
+        return best;
+    }
+
+    RHEAIndividual rankSelection()
+    {
+        population.sort(Comparator.naturalOrder());
+        int rankSum = 0;
+        for(int i = 0; i < population.size(); ++i)
+            rankSum += i + 1;
+        int ran = randomGenerator.nextInt(rankSum);
+        int p = 0;
+        for(int i = 0; i < population.size(); ++i)
+        {
+            p += population.size() - (i + 1);
+            if(p >= ran)
+                return population.get(i);
+        }
+        throw new RuntimeException("Random Generator generated an invalid goal, goal: " + ran + " p: " + p);
+    }
+    /**
+     * Run evolutionary process for one generation
+     * @param stateObs - current game state
+     */
+    private void runIteration(AbstractGameState stateObs) {
+        ElapsedCpuTimer elapsedTimerIteration = new ElapsedCpuTimer();
+
+        //selection
+        population.sort(Comparator.naturalOrder());
+
+        //copy elites
+        ArrayList<RHEAIndividual> newPopulation = new ArrayList<RHEAIndividual>();
+        int statesUpdated = 0;
+        for(int i = 0; i < params.eliteCount; ++i)
+        {
+            newPopulation.add(new RHEAIndividual(population.get(i))); // todo: possibly cheating, needs to update copy calls?
+        }
+
+        //crossover & mutation
+        for(int i = 0; i < params.childCount; ++i)
+        {
+            RHEAIndividual[] parents = selectParents();
+
+            RHEAIndividual child = crossover(parents[0], parents[1]);
+            statesUpdated += child.mutate(getForwardModel(), getPlayerID());
+            //fmCalls += child.rollout(child.gameStates[0], getForwardModel(), 0, child.actions.length, getPlayerID());
```

Comment:
My understanding of this is that we crossover the genomes (which consist of the actions), and then mutate will pick a random action index, and rollout just from that point forward - starting from the game state stored against that action. At no point (that I can see at the moment) do we rollout using all the actions in the genome, but instead assume that we can just pick up at the stored game state.

I think this may be missing a few things (or possibly I am). Consider the TicTacToe example (from comment in crossover). If we crossover two parent genomes so that we now have something like:
middle - middle - top right - ...

The state stored against the first middle is from parent 1 - and is the start state, so fine.
The state stored against the second middle is from parent 2 - and will have a cross in the middle (and a nought elsewhere. 
The score given to this sequence of actions will just depend (correct?) on the rollout from the state picked at the mutation index - which may not actually include the very first action.